Library/Homebrew/test/test_cmd_audit.rb

```diff
@@ -13,53 +13,46 @@ def teardown
     FileUtils.rm_rf @dir
   end
 
-  def formula_text(name, text)
+  def formula_text(name, body = nil, patch = nil)
     path = Pathname.new "#{@dir}/#{name}.rb"
-    path.open("w") { |f| f.write text }
+    path.open("w") do |f|
+      f.write <<-EOS.undent
+        class #{Formulary.class_s(name)} < Formula
+          #{body}
+        end
+        #{patch}
+      EOS
+    end
     FormulaText.new path
   end
 
   def test_simple_valid_formula
-    ft = formula_text "valid", <<-EOS.undent
-      class Valid < Formula
-        url "http://www.example.com/valid-1.0.tar.gz"
-      end
-    EOS
+    ft = formula_text "valid", 'url "http://www.example.com/valid-1.0.tar.gz"'
 
     refute ft.has_DATA?, "The formula doesn't have DATA"
     refute ft.has_END?, "The formula doesn't have __END__"
-    assert ft.has_trailing_newline?, "The formula have a trailing newline"
+    assert ft.has_trailing_newline?, "The formula has a trailing newline"
 
-    assert ft =~ /\burl\b/, "The formula match 'url'"
+    assert ft =~ /\burl\b/, "The formula matches 'url'"
     assert_nil ft.line_number(/desc/), "The formula doesn't match 'desc'"
     assert_equal 2, ft.line_number(/\burl\b/)
+    assert ft.include?("Valid"), "The formula includes \"Valid\""
```

Comment:
Unfortunately, this raises an error (`TypeError: no implicit conversion of FormulaText into String`). And `ft`'s `@text` can't be accessed, either.